src/number_theory/modular_forms/jacobi_theta.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import analysis.fourier.poisson_summation
+import analysis.schwartz_space
+import analysis.p_series
+import analysis.special_functions.gaussian
+
+/-! # Jacobi's theta function
+
+This file defines the Jacobi theta function
+
+$$\theta(\tau) = \sum_{n \in \mathbb{Z}} \exp (i \pi n ^ 2 \tau),$$
+
+and proves the modular transformation properties `θ (τ + 2) = θ τ` and
+`θ (-1 / τ) = (-I * τ) ^ (1 / 2) * θ τ`, using Poisson's summation formula for the latter.
+-/
+
+open filter asymptotics function set complex (hiding abs_of_nonneg sq_abs) real (hiding exp)
+
+open_locale real topology big_operators filter fourier_transform
+
+/-- Jacobi's theta function `∑' (n : ℤ), exp (π * I * n ^ 2 * τ)`. -/
+noncomputable def jacobi_theta (τ : ℂ) : ℂ := ∑' (n : ℤ), cexp (π * I * n ^ 2 * τ)
```

Comment:
It might be useful to add something like `jacobi_theta_summable` proving that this series is summable, since thats quite useful when one actually wants to work with this. It should also be really easy to prove using `summable_geometric_iff_norm_lt_1` and `complex.abs_exp`

Reply:
I added `jacobi_theta_summable`. It wasn't quite as easy as expected, maybe I was doing it in an unnecessarily painful way. (At some point we will need the stronger statement that the series is _uniformly_ convergent on suitable subsets of `ℍ`, but we can deal with that when it arises.)

Reply:
> At some point we will need the stronger statement that the series is uniformly convergent on suitable subsets of ℍ

On second thoughts, we might as well do this now, since it's not actually any harder than the pointwise summability. I've pushed the fix.